src/web/app/pages-instructor/instructor-course-edit-page/copy-instructors-from-other-courses-modal/copy-instructors-from-other-courses-modal.component.spec.ts

```diff
@@ -0,0 +1,315 @@
+import { LoadingSpinnerModule } from '../../../components/loading-spinner/loading-spinner.module';
+import { PanelChevronModule } from '../../../components/panel-chevron/panel-chevron.module';
+import { TeammatesCommonModule } from '../../../components/teammates-common/teammates-common.module';
+import { CourseTabModel, InstructorToCopyCandidateModel } from './copy-instructors-from-other-courses-modal-model';
+import { CopyInstructorsFromOtherCoursesModalComponent } from './copy-instructors-from-other-courses-modal.component';
+
+describe('CopyInstructorsFromOtherCoursesModalComponent', () => {
+
+  const testInstructor1: Instructor = {
+    googleId: 'googleIdOfIns1',
+    courseId: 'FAN0001',
+    email: 'ins1@fan.tmt',
+    isDisplayedToStudents: true,
+    displayedToStudentsAs: 'abcdefg',
+    name: 'Instructor Cat',
+    role: InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_COOWNER,
+    joinState: JoinState.JOINED,
+    key: 'ujg2l5hj5l7jrfo28fyosfklh2892',
+    institute: 'A',
+  };
+
+  const testInstructor2: Instructor = {
+    googleId: 'googleIdOfIns2',
+    courseId: 'FAN0001',
+    email: 'ins2@fan.tmt',
+    isDisplayedToStudents: false,
+    displayedToStudentsAs: '',
+    name: 'Instructor Dog',
+    role: InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_CUSTOM,
+    joinState: JoinState.JOINED,
+    key: '1343jbcl3iru2yct0897goji',
+    institute: 'A',
+  };
+
+  const testInstructor3: Instructor = {
+    googleId: 'googleIdOfIns3',
+    courseId: 'FAN0002',
+    email: 'ins1@fan.tmt',
+    isDisplayedToStudents: true,
+    displayedToStudentsAs: 'QWQWQWQ',
+    name: 'King',
+    role: InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_MANAGER,
+    joinState: JoinState.NOT_JOINED,
+    key: 'pjkjbnc523iosdk2389nfbfib2',
+    institute: 'A',
+  };
+
+  const testInstructor4: Instructor = {
+    googleId: 'googleIdOfIns4',
+    courseId: 'FAN0002',
+    email: 'ins4@fan.tmt',
+    isDisplayedToStudents: false,
+    displayedToStudentsAs: 'oldname',
+    name: 'Nothing',
+    role: InstructorPermissionRole.INSTRUCTOR_PERMISSION_ROLE_TUTOR,
+    joinState: JoinState.JOINED,
+    key: 'qdnjvbi47t928cjqnk1o8ry2o',
+    institute: 'A',
+  };
+
+  const testInstructorCandidate1: InstructorToCopyCandidateModel = {
+    instructor: testInstructor1,
+    isSelected: false,
+  };
+
+  const testInstructorCandidate2: InstructorToCopyCandidateModel = {
+    instructor: testInstructor2,
+    isSelected: false,
+  };
+
+  const testInstructorCandidate3: InstructorToCopyCandidateModel = {
+    instructor: testInstructor3,
+    isSelected: false,
+  };
+
+  const testInstructorCandidate4: InstructorToCopyCandidateModel = {
+    instructor: testInstructor4,
+    isSelected: false,
+  };
+
+  const testCourseTab1: CourseTabModel = {
+    courseId: 'FAN0002',
+    courseName: 'Test Course 1',
+    creationTimestamp: 1658797215, // Tuesday, 26 July 2022 09:00:15 GMT+08:00
```

Comment:
Instead of doing this, should define the epoch ms based on the desired time:
```ts
new Date('2022-07-26T01:00:15Z').getTime()
```

Reply:
I'll change that but I actually followed what was done in `copy-questions-from-other-sessions-modal.component.spec.ts`.